File: node/zkgroup/groups/ProfileKeyCiphertext.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import NativeImpl from '../../NativeImpl';

export default class ProfileKeyCiphertext extends ByteArray {

  static SIZE = 65;

  constructor(contents: Buffer) {
    super(contents, ProfileKeyCiphertext.SIZE, true);
    NativeImpl.ProfileKeyCiphertext_CheckValidContents(contents);
  }
}

File: java/shared/java/org/signal/libsignal/zkgroup/groups/ProfileKeyCiphertext.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.groups;

import static org.signal.libsignal.internal.FilterExceptions.filterExceptions;

import org.signal.libsignal.internal.Native;
import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;

public final class ProfileKeyCiphertext extends ByteArray {
  public ProfileKeyCiphertext(byte[] contents) throws InvalidInputException {
    super(contents);
    filterExceptions(
        InvalidInputException.class,
        () -> Native.ProfileKeyCiphertext_CheckValidContents(contents));
  }
}
